tests/test_js_syntax.py

```diff
@@ -56,3 +56,42 @@ def test_spidermonkey_warning():
     "use strict";
     var x = 0999999999999;
     """).failed()
+
+
+def test_blocks_evaluated():
+    """
+    Tests that blocks of code are actually evaluated under normal
+    circumstances.
+    """
+
+    ID = ("js", "traverser", "dangerous_global")
+
+    EVIL = "eval(evilStuff)"
+    BLOCKS = (
+        "function foo() { %s; }",
+        "function foo() { %s; yield 1; }",
+        "function foo() { yield %s; }",
+        "var foo = function () { %s; }",
+        "var foo = function () { %s; yield 1; }",
+        "function* foo() { %s; }",
+        "var foo = function* () { %s; }",
+        "var foo = function () %s",
+        "var foo = () => %s",
+        "var foo = () => { %s }",
+        "if (true) { %s }",
+        "if (true) ; else { %s }",
+        "while (true) { %s }",
+        "do { %s } while (true)",
+        "for (;;) { %s }",
+        "try { %s } catch (e) {}",
+        "try {} catch (e) { %s }",
+        "try {} finally { %s }",
+        "try {} catch (e if %s) {}",
+        # This is broken.
+        # "let (foo = bar) { %s }",
```

Comment:
It's https://bugzil.la/1009181, but support for `let` blocks has finally been removed, so I'll just remove the comment.